src/applications/static-pages/events/helpers/index.js

```diff
@@ -64,20 +64,70 @@ export const deriveMostRecentDate = (
   return futureDates[0];
 };
 
-export const removeDuplicateEvents = events => {
-  return events?.reduce((allEvents, event) => {
-    const currentEventIds = allEvents.map(
-      currentEvent => currentEvent.entityId,
-    );
+function hideEventRepeatsFromList(allEvents, event) {
```

Comment:
This reusable logic looks great! 

I'm wondering about the name of the function, though. The `reduce()` function that calls it ultimately is what "hides repeats", but this function as a standalone unit actually seems to be the inverse.  It seems this function is about including rather than hiding. It seems to me something more like, `includeItemInCollectionIfUnique(collection, item)`. 

I also wonder about the two base conditions. Might it be better to use what's below?
```
if (!allEvents || allEvents === []) {
   return [event];
}
```

Do we even need the check for `!allEvents`?

Reply:
I agree about the name being off. Something like `keepUniqueEventsFromList` sounds good to me.

I think the two base conditions will need to stay separate. The first one is to save the function from crashing if an empty events array is passed and still returns a workable data type (this came up when I was first testing it). The second one is just a time-saver for when the first event is checked - if there's no event collection to check against, it can just be added right away.

Reply:
Great!